src/util/upnp_quirks.cc

```diff
@@ -40,7 +40,7 @@
 Quirks::Quirks(const UpnpXMLBuilder& xmlBuilder, const ClientManager& clients, const std::shared_ptr<GrbNet>& addr, const std::string& userAgent)
     : xmlBuilder(xmlBuilder)
 {
-    if (addr || !userAgent.empty())
+    if (addr && !userAgent.empty())
```

Comment:
That's breaking things. getInfo has to be called if either `addr` is set or `userAgent`